string[] names
---
float32[] stiffnesses